### PR TITLE
LPD-35353 Allow numeric fields to be localized

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPD-43701](https://liferay.atlassian.net/browse/LPD-43701)

Obs.: this is sent on top of https://github.com/liferay-bpm/liferay-portal/pull/3945

[LPD-43701]: https://liferay.atlassian.net/browse/LPD-43701?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
I just rebased on top of https://github.com/brianchandotcom/liferay-portal/pull/156835

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:stop

**Comment by devpedrotavares** [comment]
Hi @marcelabc , I found an issue related to indexing for the precision decimal type. I'm working on fixing it.

**Comment by marcelabc** [comment]
Hey @devpedrotavares 
Let me know once you finish the adjustments, please :smile:

**Comment by devpedrotavares** [comment]
Hi @marcelabc , I just added one commit with the adjustments

**Comment by devpedrotavares** [comment]
Hi @marcelabc , I added a commit with this refactoring

**Comment by devpedrotavares** [comment]
Hi @marcelabc , just did a rebase on master

**Comment by devpedrotavares** [comment]
Hi @marcelabc , I just added one commit with a test for indexing localized object entries. No changes to the previous commits  😀

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -4867,25 +4866,25 @@ private void _validateListTypeEntryKey(
...
-			ObjectEntry objectEntry, long objectFieldId, long userId)
+			ObjectEntry existingObjectEntry, long listTypeDefinitionId,
+			ObjectField objectField, long userId, Serializable value)
 		throws PortalException {
 
-		Map<String, Serializable> values = objectEntry.getValues();
+		Map<String, Serializable> values = existingObjectEntry.getValues();
 
-		Serializable value = values.get(entry.getKey());
+		Serializable existingValue = values.get(objectField.getName());
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4875]: We should choose between: `values / value` OR `existingValues / existingValue`.
I would choose: `existingValues / existingValue`.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 1 above)
Comment (devpedrotavares) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: @marcelabc Done!

**Comment by devpedrotavares** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:object

**Comment by devpedrotavares** [comment]
ci:report:90325019

**Comment by devpedrotavares** [comment]
ci:report:90325023

**Comment by devpedrotavares** [comment]
Just added a commit adding the feature flag to the test classes

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:object

**Comment by devpedrotavares** [comment]
ci:forward:force

**Pull request closed without merging**